test/common/testlib.py

```diff
@@ -1006,6 +1032,7 @@ def assert_pixels_in_current_layout(self, selector: str, key: str,
         if not (Image and self.pixels_label):
             return
 
+        self._set_rtl("rtl" in self.current_layout["name"])
```

Comment:
I don't like that we do this for *every* scenario. It's expensive to do, but moreover it changes the pages, and thus we don't validate the default layout any more. I. e. after this we could accidentally enable RTL for English and still get passing tests.

Could this move into assert_pixels(), and only switch once it reaches the "rtl" scenario? Then of course the layout list needs a comment to say that all RTL scenarios must come at the end, and assert_pixels() could even grow an assertion for this (although it should be obvious enough if we screw this up, as all screenshots would yell at us). After that, set it back to LTR.

Reply:
> I don't like that we do this for _every_ scenario. It's expensive to do, but moreover it changes the pages, and thus we don't validate the default layout any more. I. e. after this we could accidentally enable RTL for English and still get passing tests.
> 
> Could this move into assert_pixels(), and only switch once it reaches the "rtl" scenario? Then of course the layout list needs a comment to say that all RTL scenarios must come at the end, and assert_pixels() could even grow an assertion for this (although it should be obvious enough if we screw this up, as all screenshots would yell at us). After that, set it back to LTR.

I don't see how this will work. Even if we have  the RTL scenario in the end of the list, we want to undo the RTL setting after the assert pixels for the RTL scenario. This is because test continues to run after the assert_pixels call , and there might be a followup assert_pixels in the same test or the test might fail, which generates a screenshot, and we want that to be the original screen, not RT.

So I adjust the code to call set_rtl only on the RTL scenario, but we definitely need to revert this after this scenario.

Reply:
Yes, of course -- that's what I meant with "After that, set it back to LTR." There's already some reset code after the loop which iterates the scenarios. It's still DOM hacking (which is super ugly), but I don't think we can avoid it with our current language switching. This whole "RTL scenario" is a big hammer..